analyzer/emerald/incoming.go

```diff
@@ -334,21 +335,25 @@ func extractRound(sigContext signature.Context, b *block.Block, txrs []*sdkClien
 		if !foundGasUsedEvent {
 			if (txr.Result.IsUnknown() || txr.Result.IsSuccess()) && tx != nil {
 				// Treat as if it used all the gas.
-				txGasUsed = int64(tx.AuthInfo.Fee.Gas)
+				txGasUsed = tx.AuthInfo.Fee.Gas
 			} else { //nolint:staticcheck
 				// Inaccurate: Treat as not using any gas.
 			}
 		}
 		blockData.TransactionData = append(blockData.TransactionData, &blockTransactionData)
-		blockData.GasUsed += txGasUsed
+		if newGasUsed, carry := bits.Add64(blockData.GasUsed, txGasUsed, 0); carry == 0 {
+			blockData.GasUsed = newGasUsed
+		} else {
+			return nil, fmt.Errorf("gas used overflow")
+		}
```

Comment:
Nit: Given that we error out, this can be just
```suggestion
		if blockData.GasUsed, carry := bits.Add64(blockData.GasUsed, txGasUsed, 0); carry != 0 {
			return nil, fmt.Errorf("gas used overflow")
		}
```
probably?

I don't know how (im)possible this overflow scenario is. Do you have ballpark credible intervals for gas per tx and txs per block?
If we think it's at all possible, erroring out (and blocking the indexer from ever progressing past this block) will be very unpleasant to firefight when it happens. Instead, we could
- use bigint for this field (in db and in go), or
- cap the sum at `math.MaxUint64`. We could even note it in the API docs for the pedants.
- ignore the problem, as you suggest between the lines. Depends on how impossible the overflow is.

I expect the second option is more ergonomic for almost every client. If anybody _really_ cares about the real sum (I don't see why they would, "lotsa gas" is probably close enough), they can detect the MAX_UINT value and fetch txs for that block one by one to derive the real total gas cost.

Reply:
re: 3-line version, I was thinking that too but

```
$ go build .
# github.com/oasisprotocol/oasis-indexer/analyzer/emerald
analyzer/emerald/incoming.go:344:6: non-name blockData.GasUsed on left side of :=
```

Reply:
it's in the realm of "this won't be the only thing that goes wrong if it happens." the runtime sdk uses u64 for max batch gas

Reply:
Oh that's very useful! I didn't think of that parameter. Then I suggest we simply not check for overflow here, and just note `// guaranteed to not overflow; max total batch gas is defined as u64 in oasis-sdk`

Reply:
removed!

Reply:
Thank you!